[2025-04-25T11:53:12.482+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: traffic_data_pipeline.submit_traffic_processor_spark_job manual__2025-04-25T11:53:10.443602+00:00 [queued]>
[2025-04-25T11:53:12.506+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: traffic_data_pipeline.submit_traffic_processor_spark_job manual__2025-04-25T11:53:10.443602+00:00 [queued]>
[2025-04-25T11:53:12.508+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2025-04-25T11:53:12.550+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): submit_traffic_processor_spark_job> on 2025-04-25 11:53:10.443602+00:00
[2025-04-25T11:53:12.564+0000] {standard_task_runner.py:60} INFO - Started process 419 to run task
[2025-04-25T11:53:12.572+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'traffic_data_pipeline', 'submit_traffic_processor_spark_job', 'manual__2025-04-25T11:53:10.443602+00:00', '--job-id', '17', '--raw', '--subdir', 'DAGS_FOLDER/traffic_pipeline_dag.py', '--cfg-path', '/tmp/tmp8n7m_twc']
[2025-04-25T11:53:12.577+0000] {standard_task_runner.py:88} INFO - Job 17: Subtask submit_traffic_processor_spark_job
[2025-04-25T11:53:12.649+0000] {task_command.py:423} INFO - Running <TaskInstance: traffic_data_pipeline.submit_traffic_processor_spark_job manual__2025-04-25T11:53:10.443602+00:00 [running]> on host 237827c6637e
[2025-04-25T11:53:12.929+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='traffic_data_pipeline' AIRFLOW_CTX_TASK_ID='submit_traffic_processor_spark_job' AIRFLOW_CTX_EXECUTION_DATE='2025-04-25T11:53:10.443602+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-25T11:53:10.443602+00:00'
[2025-04-25T11:53:12.931+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-04-25T11:53:12.933+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\n            export KAFKA_BROKERS_INTERNAL="kafk-1:9092,kafk-2:9092,kafk-3:9092" && \\\n            export KAFKA_TOPIC="raw_traffic_data" && \\\n            export SCHEMA_REGISTRY_URL="http://schema-registry:8081" && \\\n            export REDIS_HOST="redis" && \\\n            export CLICKHOUSE_HOST="clickhouse-server" && \\\n            export CLICKHOUSE_NATIVE_PORT="9000" && \\\n            export MINIO_ENDPOINT="http://minio:9000" && \\\n            export MINIO_ACCESS_KEY="minioadmin" && \\\n            export MINIO_SECRET_KEY="minioadmin" && \\\n            export MINIO_BUCKET="traffic-data" && \\\n            export CHECKPOINT_LOCATION="s3a://traffic-data/checkpoints_***" && \\\n            export PYSPARK_PYTHON=/usr/bin/python3 && \\\n            spark-submit \\\n            --master local[*] \\\n            --deploy-mode client \\\n            --conf spark.sql.streaming.schemaInference=true \\\n            --conf spark.hadoop.fs.s3a.endpoint=http://minio:9000 \\\n            --conf spark.hadoop.fs.s3a.access.key=minioadmin \\\n            --conf spark.hadoop.fs.s3a.secret.key=minioadmin \\\n            --conf spark.hadoop.fs.s3a.path.style.access=true \\\n            --conf spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem \\\n            --conf spark.sql.avro.schemaRegistryUrl=http://schema-registry:8081 \\\n            /opt/spark/app/processor.py\n            ']
[2025-04-25T11:53:12.945+0000] {subprocess.py:86} INFO - Output:
[2025-04-25T11:53:12.955+0000] {subprocess.py:93} INFO - /opt/spark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-04-25T11:53:16.031+0000] {subprocess.py:93} INFO - --- Starting Spark Streaming Traffic Processor ---
[2025-04-25T11:53:16.032+0000] {subprocess.py:93} INFO - Reading from Kafka: kafk-1:9092,kafk-2:9092,kafk-3:9092 / Topic: raw_traffic_data
[2025-04-25T11:53:16.033+0000] {subprocess.py:93} INFO - Schema Registry URL: http://schema-registry:8081
[2025-04-25T11:53:16.033+0000] {subprocess.py:93} INFO - Writing state to Redis: redis:6379
[2025-04-25T11:53:16.034+0000] {subprocess.py:93} INFO - Writing history to ClickHouse: clickhouse-server:9000 (DB: traffic_db)
[2025-04-25T11:53:16.035+0000] {subprocess.py:93} INFO - Writing Parquet to MinIO Bucket: traffic-data
[2025-04-25T11:53:16.035+0000] {subprocess.py:93} INFO - Checkpoint Location Base: s3a://traffic-data/checkpoints_***
[2025-04-25T11:53:16.118+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO SparkContext: Running Spark version 3.3.4
[2025-04-25T11:53:16.177+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-04-25T11:53:16.311+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO ResourceUtils: ==============================================================
[2025-04-25T11:53:16.312+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-04-25T11:53:16.313+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO ResourceUtils: ==============================================================
[2025-04-25T11:53:16.313+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO SparkContext: Submitted application: TrafficProcessor
[2025-04-25T11:53:16.357+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-04-25T11:53:16.367+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO ResourceProfile: Limiting resource is cpu
[2025-04-25T11:53:16.368+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-04-25T11:53:16.416+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO SecurityManager: Changing view acls to: ***
[2025-04-25T11:53:16.417+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO SecurityManager: Changing modify acls to: ***
[2025-04-25T11:53:16.418+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO SecurityManager: Changing view acls groups to:
[2025-04-25T11:53:16.419+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO SecurityManager: Changing modify acls groups to:
[2025-04-25T11:53:16.419+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(***); groups with view permissions: Set(); users  with modify permissions: Set(***); groups with modify permissions: Set()
[2025-04-25T11:53:16.618+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO Utils: Successfully started service 'sparkDriver' on port 46449.
[2025-04-25T11:53:16.648+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO SparkEnv: Registering MapOutputTracker
[2025-04-25T11:53:16.690+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO SparkEnv: Registering BlockManagerMaster
[2025-04-25T11:53:16.747+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-04-25T11:53:16.760+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-04-25T11:53:16.767+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-04-25T11:53:16.819+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-9d1b0430-15f9-4d41-82a9-e7ba68d90b3f
[2025-04-25T11:53:16.856+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-04-25T11:53:16.912+0000] {subprocess.py:93} INFO - 25/04/25 11:53:16 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-04-25T11:53:17.185+0000] {subprocess.py:93} INFO - 25/04/25 11:53:17 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-04-25T11:53:17.313+0000] {subprocess.py:93} INFO - 25/04/25 11:53:17 INFO Executor: Starting executor ID driver on host 237827c6637e
[2025-04-25T11:53:17.326+0000] {subprocess.py:93} INFO - 25/04/25 11:53:17 INFO Executor: Starting executor with user classpath (userClassPathFirst = false): ''
[2025-04-25T11:53:17.358+0000] {subprocess.py:93} INFO - 25/04/25 11:53:17 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 45995.
[2025-04-25T11:53:17.360+0000] {subprocess.py:93} INFO - 25/04/25 11:53:17 INFO NettyBlockTransferService: Server created on 237827c6637e:45995
[2025-04-25T11:53:17.362+0000] {subprocess.py:93} INFO - 25/04/25 11:53:17 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-04-25T11:53:17.380+0000] {subprocess.py:93} INFO - 25/04/25 11:53:17 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 237827c6637e, 45995, None)
[2025-04-25T11:53:17.385+0000] {subprocess.py:93} INFO - 25/04/25 11:53:17 INFO BlockManagerMasterEndpoint: Registering block manager 237827c6637e:45995 with 434.4 MiB RAM, BlockManagerId(driver, 237827c6637e, 45995, None)
[2025-04-25T11:53:17.388+0000] {subprocess.py:93} INFO - 25/04/25 11:53:17 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 237827c6637e, 45995, None)
[2025-04-25T11:53:17.390+0000] {subprocess.py:93} INFO - 25/04/25 11:53:17 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 237827c6637e, 45995, None)
[2025-04-25T11:53:17.838+0000] {subprocess.py:93} INFO - Spark Session created. Version: 3.3.4
[2025-04-25T11:53:17.839+0000] {subprocess.py:93} INFO - Broadcasting segments data...
[2025-04-25T11:53:17.960+0000] {subprocess.py:93} INFO - Segments data broadcasted.
[2025-04-25T11:53:20.593+0000] {subprocess.py:93} INFO - Reading Avro schema from file: /opt/spark/schemas/raw_traffic_event.avsc
[2025-04-25T11:53:20.596+0000] {subprocess.py:93} INFO - Successfully read Avro schema string.
[2025-04-25T11:53:20.597+0000] {subprocess.py:93} INFO - Deserializing Kafka messages using schema string from file...
[2025-04-25T11:53:20.746+0000] {subprocess.py:93} INFO - 25/04/25 11:53:20 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:21.203+0000] {subprocess.py:93} INFO - Using checkpoint location: s3a://traffic-data/checkpoints_***/traffic_processing_pipeline
[2025-04-25T11:53:21.378+0000] {subprocess.py:93} INFO - 25/04/25 11:53:21 WARN MetricsConfig: Cannot locate configuration: tried hadoop-metrics2-s3a-file-system.properties,hadoop-metrics2.properties
[2025-04-25T11:53:22.069+0000] {subprocess.py:93} INFO - 25/04/25 11:53:22 WARN ResolveWriteToStream: spark.sql.adaptive.enabled is not supported in streaming DataFrames/Datasets and will be disabled.
[2025-04-25T11:53:22.269+0000] {subprocess.py:93} INFO - Streaming query 'traffic_processing_pipeline' started.
[2025-04-25T11:53:23.069+0000] {subprocess.py:93} INFO - 25/04/25 11:53:23 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:23.159+0000] {subprocess.py:93} INFO - 25/04/25 11:53:23 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:23.747+0000] {subprocess.py:93} INFO - 25/04/25 11:53:23 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:23.767+0000] {subprocess.py:93} INFO - 25/04/25 11:53:23 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:23.770+0000] {subprocess.py:93} INFO - 25/04/25 11:53:23 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:23.772+0000] {subprocess.py:93} INFO - 25/04/25 11:53:23 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:24.385+0000] {subprocess.py:93} INFO - 25/04/25 11:53:24 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:24.385+0000] {subprocess.py:93} INFO - 25/04/25 11:53:24 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:24.395+0000] {subprocess.py:93} INFO - 25/04/25 11:53:24 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:24.396+0000] {subprocess.py:93} INFO - 25/04/25 11:53:24 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:24.931+0000] {subprocess.py:93} INFO - 25/04/25 11:53:24 WARN KafkaDataConsumer: KafkaDataConsumer is not running in UninterruptibleThread. It may hang when KafkaDataConsumer's methods are interrupted because of KAFKA-1894
[2025-04-25T11:53:24.932+0000] {subprocess.py:93} INFO - 25/04/25 11:53:24 WARN KafkaDataConsumer: KafkaDataConsumer is not running in UninterruptibleThread. It may hang when KafkaDataConsumer's methods are interrupted because of KAFKA-1894
[2025-04-25T11:53:25.632+0000] {subprocess.py:93} INFO - 25/04/25 11:53:25 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:25.649+0000] {subprocess.py:93} INFO - 25/04/25 11:53:25 ERROR PythonUDFRunner: Python worker exited unexpectedly (crashed)
[2025-04-25T11:53:25.650+0000] {subprocess.py:93} INFO - org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-25T11:53:25.651+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 666, in main
[2025-04-25T11:53:25.651+0000] {subprocess.py:93} INFO -     eval_type = read_int(infile)
[2025-04-25T11:53:25.652+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^
[2025-04-25T11:53:25.653+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 595, in read_int
[2025-04-25T11:53:25.654+0000] {subprocess.py:93} INFO -     raise EOFError
[2025-04-25T11:53:25.655+0000] {subprocess.py:93} INFO - EOFError
[2025-04-25T11:53:25.655+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:53:25.656+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-25T11:53:25.656+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-25T11:53:25.657+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-25T11:53:25.658+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-25T11:53:25.658+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-25T11:53:25.659+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-25T11:53:25.659+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.660+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.660+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-04-25T11:53:25.661+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:53:25.662+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:53:25.662+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.663+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:53:25.663+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.664+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.665+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:53:25.665+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:53:25.666+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.667+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:53:25.667+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:53:25.668+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:53:25.668+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:53:25.669+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:53:25.669+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:53:25.670+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:53:25.670+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:53:25.671+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:53:25.671+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:53:25.672+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:53:25.672+0000] {subprocess.py:93} INFO - 	... 17 more
[2025-04-25T11:53:25.673+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:53:25.673+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:53:25.674+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:53:25.674+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:53:25.675+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.676+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:53:25.677+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:53:25.677+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:53:25.678+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.678+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:53:25.679+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:53:25.679+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:53:25.680+0000] {subprocess.py:93} INFO - 25/04/25 11:53:25 ERROR PythonUDFRunner: This may have been caused by a prior exception:
[2025-04-25T11:53:25.680+0000] {subprocess.py:93} INFO - org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:53:25.681+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:53:25.682+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:53:25.682+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:53:25.683+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:53:25.684+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.684+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:53:25.685+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.685+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.686+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:53:25.686+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:53:25.687+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.688+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:53:25.688+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:53:25.689+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:53:25.690+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:53:25.690+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:53:25.691+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:53:25.691+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:53:25.692+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:53:25.693+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:53:25.693+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:53:25.694+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:53:25.694+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:53:25.695+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.696+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:53:25.697+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:53:25.697+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:53:25.698+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.699+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:53:25.699+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:53:25.700+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:53:25.701+0000] {subprocess.py:93} INFO - 25/04/25 11:53:25 ERROR PythonUDFRunner: Python worker exited unexpectedly (crashed)
[2025-04-25T11:53:25.702+0000] {subprocess.py:93} INFO - org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-25T11:53:25.702+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 666, in main
[2025-04-25T11:53:25.703+0000] {subprocess.py:93} INFO -     eval_type = read_int(infile)
[2025-04-25T11:53:25.704+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^
[2025-04-25T11:53:25.704+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 595, in read_int
[2025-04-25T11:53:25.705+0000] {subprocess.py:93} INFO -     raise EOFError
[2025-04-25T11:53:25.705+0000] {subprocess.py:93} INFO - EOFError
[2025-04-25T11:53:25.706+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:53:25.706+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-25T11:53:25.707+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-25T11:53:25.708+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-25T11:53:25.708+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-25T11:53:25.709+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-25T11:53:25.709+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-25T11:53:25.710+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.711+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.712+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage3.processNext(Unknown Source)
[2025-04-25T11:53:25.712+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:53:25.713+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:53:25.713+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.714+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage4.hashAgg_doAggregateWithKeys_0$(Unknown Source)
[2025-04-25T11:53:25.714+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage4.processNext(Unknown Source)
[2025-04-25T11:53:25.714+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:53:25.715+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:53:25.715+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.716+0000] {subprocess.py:93} INFO - 	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:140)
[2025-04-25T11:53:25.716+0000] {subprocess.py:93} INFO - 	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
[2025-04-25T11:53:25.717+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
[2025-04-25T11:53:25.717+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
[2025-04-25T11:53:25.718+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.Task.run(Task.scala:136)
[2025-04-25T11:53:25.718+0000] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:548)
[2025-04-25T11:53:25.718+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1504)
[2025-04-25T11:53:25.719+0000] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:551)
[2025-04-25T11:53:25.719+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2025-04-25T11:53:25.719+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2025-04-25T11:53:25.720+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2025-04-25T11:53:25.720+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:53:25.721+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:53:25.721+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:53:25.721+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:53:25.722+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:53:25.722+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.723+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:53:25.724+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.724+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.725+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:53:25.725+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:53:25.725+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.726+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:53:25.726+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:53:25.727+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:53:25.727+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:53:25.727+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:53:25.728+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:53:25.728+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:53:25.729+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:53:25.729+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:53:25.730+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:53:25.730+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:53:25.731+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:53:25.731+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.732+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:53:25.732+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:53:25.733+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:53:25.733+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.733+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:53:25.734+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:53:25.734+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:53:25.735+0000] {subprocess.py:93} INFO - 25/04/25 11:53:25 ERROR PythonUDFRunner: This may have been caused by a prior exception:
[2025-04-25T11:53:25.735+0000] {subprocess.py:93} INFO - org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:53:25.735+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:53:25.736+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:53:25.736+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:53:25.737+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:53:25.737+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.738+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:53:25.738+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.738+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.739+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:53:25.740+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:53:25.740+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.741+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:53:25.741+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:53:25.741+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:53:25.742+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:53:25.742+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:53:25.743+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:53:25.743+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:53:25.744+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:53:25.744+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:53:25.745+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:53:25.745+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:53:25.746+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:53:25.746+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.746+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:53:25.747+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:53:25.747+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:53:25.748+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.748+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:53:25.748+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:53:25.749+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:53:25.750+0000] {subprocess.py:93} INFO - 25/04/25 11:53:25 ERROR Executor: Exception in task 1.0 in stage 0.0 (TID 1)
[2025-04-25T11:53:25.750+0000] {subprocess.py:93} INFO - org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:53:25.751+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:53:25.751+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:53:25.752+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:53:25.752+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:53:25.752+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.753+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:53:25.753+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.754+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.754+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:53:25.755+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:53:25.755+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.756+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:53:25.756+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:53:25.757+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:53:25.757+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:53:25.758+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:53:25.758+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:53:25.759+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:53:25.759+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:53:25.760+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:53:25.760+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:53:25.761+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:53:25.761+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:53:25.762+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.762+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:53:25.763+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:53:25.763+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:53:25.764+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.764+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:53:25.765+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:53:25.766+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:53:25.766+0000] {subprocess.py:93} INFO - 25/04/25 11:53:25 WARN TaskSetManager: Lost task 1.0 in stage 0.0 (TID 1) (237827c6637e executor driver): org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:53:25.766+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:53:25.767+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:53:25.768+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:53:25.768+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:53:25.769+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.769+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:53:25.770+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.771+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.771+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:53:25.772+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:53:25.773+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.773+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:53:25.774+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:53:25.775+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:53:25.775+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:53:25.776+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:53:25.776+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:53:25.777+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:53:25.777+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:53:25.778+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:53:25.779+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:53:25.779+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:53:25.780+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:53:25.781+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.781+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:53:25.782+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:53:25.783+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:53:25.783+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.784+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:53:25.785+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:53:25.785+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:53:25.786+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:53:25.787+0000] {subprocess.py:93} INFO - 25/04/25 11:53:25 ERROR TaskSetManager: Task 1 in stage 0.0 failed 1 times; aborting job
[2025-04-25T11:53:25.856+0000] {subprocess.py:93} INFO - 25/04/25 11:53:25 ERROR MicroBatchExecution: Query traffic_processing_pipeline [id = 5e03d62e-cf04-4af4-95a7-b2203ba8cf76, runId = 49dcd9a7-b550-488f-bc81-b7d7e1831429] terminated with error
[2025-04-25T11:53:25.857+0000] {subprocess.py:93} INFO - py4j.Py4JException: An exception was raised by the Python Proxy. Return Message: Traceback (most recent call last):
[2025-04-25T11:53:25.858+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/clientserver.py", line 617, in _call_proxy
[2025-04-25T11:53:25.858+0000] {subprocess.py:93} INFO -     return_value = getattr(self.pool[obj_id], method)(*params)
[2025-04-25T11:53:25.859+0000] {subprocess.py:93} INFO -                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-25T11:53:25.860+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 276, in call
[2025-04-25T11:53:25.860+0000] {subprocess.py:93} INFO -     raise e
[2025-04-25T11:53:25.861+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 273, in call
[2025-04-25T11:53:25.861+0000] {subprocess.py:93} INFO -     self.func(DataFrame(jdf, wrapped_session_jdf), batch_id)
[2025-04-25T11:53:25.862+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 292, in <lambda>
[2025-04-25T11:53:25.863+0000] {subprocess.py:93} INFO -     write_to_redis(df, epoch_id),
[2025-04-25T11:53:25.863+0000] {subprocess.py:93} INFO -     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-25T11:53:25.864+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 64, in write_to_redis
[2025-04-25T11:53:25.865+0000] {subprocess.py:93} INFO -     if df.isEmpty(): return
[2025-04-25T11:53:25.865+0000] {subprocess.py:93} INFO -        ^^^^^^^^^^^^
[2025-04-25T11:53:25.866+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/dataframe.py", line 553, in isEmpty
[2025-04-25T11:53:25.866+0000] {subprocess.py:93} INFO -     return self._jdf.isEmpty()
[2025-04-25T11:53:25.867+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^
[2025-04-25T11:53:25.867+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py", line 1321, in __call__
[2025-04-25T11:53:25.868+0000] {subprocess.py:93} INFO -     return_value = get_return_value(
[2025-04-25T11:53:25.868+0000] {subprocess.py:93} INFO -                    ^^^^^^^^^^^^^^^^^
[2025-04-25T11:53:25.869+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 190, in deco
[2025-04-25T11:53:25.870+0000] {subprocess.py:93} INFO -     return f(*a, **kw)
[2025-04-25T11:53:25.870+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^
[2025-04-25T11:53:25.871+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/protocol.py", line 326, in get_return_value
[2025-04-25T11:53:25.871+0000] {subprocess.py:93} INFO -     raise Py4JJavaError(
[2025-04-25T11:53:25.872+0000] {subprocess.py:93} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o181.isEmpty.
[2025-04-25T11:53:25.873+0000] {subprocess.py:93} INFO - : org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 0.0 failed 1 times, most recent failure: Lost task 1.0 in stage 0.0 (TID 1) (237827c6637e executor driver): org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:53:25.873+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:53:25.874+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:53:25.875+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:53:25.875+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:53:25.876+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.876+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:53:25.877+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.877+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.878+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:53:25.878+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:53:25.879+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.879+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:53:25.880+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:53:25.880+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:53:25.881+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:53:25.881+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:53:25.882+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:53:25.882+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:53:25.883+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:53:25.883+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:53:25.884+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:53:25.885+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:53:25.885+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:53:25.886+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.886+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:53:25.887+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:53:25.888+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:53:25.888+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.889+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:53:25.889+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:53:25.890+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:53:25.891+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:53:25.891+0000] {subprocess.py:93} INFO - Driver stacktrace:
[2025-04-25T11:53:25.892+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2668)
[2025-04-25T11:53:25.892+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2604)
[2025-04-25T11:53:25.893+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2603)
[2025-04-25T11:53:25.894+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-04-25T11:53:25.894+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-04-25T11:53:25.895+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-04-25T11:53:25.895+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2603)
[2025-04-25T11:53:25.895+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1178)
[2025-04-25T11:53:25.896+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1178)
[2025-04-25T11:53:25.896+0000] {subprocess.py:93} INFO - 	at scala.Option.foreach(Option.scala:407)
[2025-04-25T11:53:25.897+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1178)
[2025-04-25T11:53:25.897+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2856)
[2025-04-25T11:53:25.898+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2798)
[2025-04-25T11:53:25.898+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2787)
[2025-04-25T11:53:25.898+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2025-04-25T11:53:25.899+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:952)
[2025-04-25T11:53:25.899+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2238)
[2025-04-25T11:53:25.899+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2259)
[2025-04-25T11:53:25.900+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2278)
[2025-04-25T11:53:25.900+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:506)
[2025-04-25T11:53:25.900+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:459)
[2025-04-25T11:53:25.901+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$isEmpty$1(Dataset.scala:604)
[2025-04-25T11:53:25.901+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$isEmpty$1$adapted(Dataset.scala:603)
[2025-04-25T11:53:25.901+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:3898)
[2025-04-25T11:53:25.902+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:510)
[2025-04-25T11:53:25.902+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3896)
[2025-04-25T11:53:25.903+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)
[2025-04-25T11:53:25.903+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)
[2025-04-25T11:53:25.903+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[2025-04-25T11:53:25.904+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:53:25.904+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[2025-04-25T11:53:25.904+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3896)
[2025-04-25T11:53:25.905+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.isEmpty(Dataset.scala:603)
[2025-04-25T11:53:25.905+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-04-25T11:53:25.906+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-04-25T11:53:25.906+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-04-25T11:53:25.907+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-04-25T11:53:25.908+0000] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-04-25T11:53:25.909+0000] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2025-04-25T11:53:25.910+0000] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:282)
[2025-04-25T11:53:25.910+0000] {subprocess.py:93} INFO - 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-04-25T11:53:25.911+0000] {subprocess.py:93} INFO - 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-04-25T11:53:25.911+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.sendCommand(ClientServerConnection.java:244)
[2025-04-25T11:53:25.911+0000] {subprocess.py:93} INFO - 	at py4j.CallbackClient.sendCommand(CallbackClient.java:384)
[2025-04-25T11:53:25.912+0000] {subprocess.py:93} INFO - 	at py4j.CallbackClient.sendCommand(CallbackClient.java:356)
[2025-04-25T11:53:25.912+0000] {subprocess.py:93} INFO - 	at py4j.reflection.PythonProxyHandler.invoke(PythonProxyHandler.java:106)
[2025-04-25T11:53:25.912+0000] {subprocess.py:93} INFO - 	at jdk.proxy3/jdk.proxy3.$Proxy37.call(Unknown Source)
[2025-04-25T11:53:25.913+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1(ForeachBatchSink.scala:51)
[2025-04-25T11:53:25.913+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1$adapted(ForeachBatchSink.scala:51)
[2025-04-25T11:53:25.913+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.ForeachBatchSink.addBatch(ForeachBatchSink.scala:32)
[2025-04-25T11:53:25.914+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$17(MicroBatchExecution.scala:665)
[2025-04-25T11:53:25.914+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)
[2025-04-25T11:53:25.914+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)
[2025-04-25T11:53:25.915+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[2025-04-25T11:53:25.916+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:53:25.916+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[2025-04-25T11:53:25.917+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$16(MicroBatchExecution.scala:663)
[2025-04-25T11:53:25.917+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-25T11:53:25.918+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-25T11:53:25.918+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-25T11:53:25.918+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runBatch(MicroBatchExecution.scala:663)
[2025-04-25T11:53:25.919+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$2(MicroBatchExecution.scala:256)
[2025-04-25T11:53:25.919+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-25T11:53:25.920+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-25T11:53:25.920+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-25T11:53:25.921+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-25T11:53:25.921+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$1(MicroBatchExecution.scala:219)
[2025-04-25T11:53:25.921+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProcessingTimeExecutor.execute(TriggerExecutor.scala:67)
[2025-04-25T11:53:25.922+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runActivatedStream(MicroBatchExecution.scala:213)
[2025-04-25T11:53:25.922+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.$anonfun$runStream$1(StreamExecution.scala:307)
[2025-04-25T11:53:25.922+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-25T11:53:25.923+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:53:25.923+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.org$apache$spark$sql$execution$streaming$StreamExecution$$runStream(StreamExecution.scala:285)
[2025-04-25T11:53:25.923+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution$$anon$1.run(StreamExecution.scala:208)
[2025-04-25T11:53:25.924+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:53:25.924+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:53:25.925+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:53:25.925+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:53:25.926+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:53:25.926+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.927+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:53:25.927+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.928+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.928+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:53:25.928+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:53:25.929+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:25.929+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:53:25.929+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:53:25.930+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:53:25.930+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:53:25.931+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:53:25.931+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:53:25.931+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:53:25.932+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:53:25.932+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:53:25.933+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:53:25.933+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:53:25.933+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:53:25.934+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.934+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:53:25.935+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:53:25.935+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:53:25.935+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:25.936+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:53:25.936+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:53:25.937+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:53:25.937+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:53:25.938+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:53:25.939+0000] {subprocess.py:93} INFO - 	at py4j.Protocol.getReturnValue(Protocol.java:476)
[2025-04-25T11:53:25.939+0000] {subprocess.py:93} INFO - 	at py4j.reflection.PythonProxyHandler.invoke(PythonProxyHandler.java:108)
[2025-04-25T11:53:25.939+0000] {subprocess.py:93} INFO - 	at jdk.proxy3/jdk.proxy3.$Proxy37.call(Unknown Source)
[2025-04-25T11:53:25.940+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1(ForeachBatchSink.scala:51)
[2025-04-25T11:53:25.940+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1$adapted(ForeachBatchSink.scala:51)
[2025-04-25T11:53:25.941+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.ForeachBatchSink.addBatch(ForeachBatchSink.scala:32)
[2025-04-25T11:53:25.942+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$17(MicroBatchExecution.scala:665)
[2025-04-25T11:53:25.942+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)
[2025-04-25T11:53:25.943+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)
[2025-04-25T11:53:25.943+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[2025-04-25T11:53:25.944+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:53:25.944+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[2025-04-25T11:53:25.945+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$16(MicroBatchExecution.scala:663)
[2025-04-25T11:53:25.946+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-25T11:53:25.946+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-25T11:53:25.947+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-25T11:53:25.947+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runBatch(MicroBatchExecution.scala:663)
[2025-04-25T11:53:25.948+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$2(MicroBatchExecution.scala:256)
[2025-04-25T11:53:25.948+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-25T11:53:25.949+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-25T11:53:25.950+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-25T11:53:25.950+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-25T11:53:25.951+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$1(MicroBatchExecution.scala:219)
[2025-04-25T11:53:25.952+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProcessingTimeExecutor.execute(TriggerExecutor.scala:67)
[2025-04-25T11:53:25.952+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runActivatedStream(MicroBatchExecution.scala:213)
[2025-04-25T11:53:25.953+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.$anonfun$runStream$1(StreamExecution.scala:307)
[2025-04-25T11:53:25.953+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-25T11:53:25.954+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:53:25.955+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.org$apache$spark$sql$execution$streaming$StreamExecution$$runStream(StreamExecution.scala:285)
[2025-04-25T11:53:25.956+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution$$anon$1.run(StreamExecution.scala:208)
[2025-04-25T11:53:26.004+0000] {subprocess.py:93} INFO - 25/04/25 11:53:26 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:53:26.016+0000] {subprocess.py:93} INFO - 25/04/25 11:53:26 WARN TaskSetManager: Lost task 0.0 in stage 0.0 (TID 0) (237827c6637e executor driver): TaskKilled (Stage cancelled)
[2025-04-25T11:53:26.077+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-04-25T11:53:26.078+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 306, in <module>
[2025-04-25T11:53:26.083+0000] {subprocess.py:93} INFO -     main()
[2025-04-25T11:53:26.084+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 299, in main
[2025-04-25T11:53:26.086+0000] {subprocess.py:93} INFO -     streaming_query.awaitTermination()
[2025-04-25T11:53:26.087+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/streaming.py", line 107, in awaitTermination
[2025-04-25T11:53:26.087+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py", line 1321, in __call__
[2025-04-25T11:53:26.088+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 196, in deco
[2025-04-25T11:53:26.094+0000] {subprocess.py:93} INFO - pyspark.sql.utils.StreamingQueryException: Query traffic_processing_pipeline [id = 5e03d62e-cf04-4af4-95a7-b2203ba8cf76, runId = 49dcd9a7-b550-488f-bc81-b7d7e1831429] terminated with exception: An exception was raised by the Python Proxy. Return Message: Traceback (most recent call last):
[2025-04-25T11:53:26.095+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/clientserver.py", line 617, in _call_proxy
[2025-04-25T11:53:26.096+0000] {subprocess.py:93} INFO -     return_value = getattr(self.pool[obj_id], method)(*params)
[2025-04-25T11:53:26.097+0000] {subprocess.py:93} INFO -                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-25T11:53:26.097+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 276, in call
[2025-04-25T11:53:26.098+0000] {subprocess.py:93} INFO -     raise e
[2025-04-25T11:53:26.098+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 273, in call
[2025-04-25T11:53:26.099+0000] {subprocess.py:93} INFO -     self.func(DataFrame(jdf, wrapped_session_jdf), batch_id)
[2025-04-25T11:53:26.099+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 292, in <lambda>
[2025-04-25T11:53:26.099+0000] {subprocess.py:93} INFO -     write_to_redis(df, epoch_id),
[2025-04-25T11:53:26.100+0000] {subprocess.py:93} INFO -     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-25T11:53:26.100+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 64, in write_to_redis
[2025-04-25T11:53:26.100+0000] {subprocess.py:93} INFO -     if df.isEmpty(): return
[2025-04-25T11:53:26.101+0000] {subprocess.py:93} INFO -        ^^^^^^^^^^^^
[2025-04-25T11:53:26.101+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/dataframe.py", line 553, in isEmpty
[2025-04-25T11:53:26.102+0000] {subprocess.py:93} INFO -     return self._jdf.isEmpty()
[2025-04-25T11:53:26.102+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^
[2025-04-25T11:53:26.102+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py", line 1321, in __call__
[2025-04-25T11:53:26.103+0000] {subprocess.py:93} INFO -     return_value = get_return_value(
[2025-04-25T11:53:26.103+0000] {subprocess.py:93} INFO -                    ^^^^^^^^^^^^^^^^^
[2025-04-25T11:53:26.104+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 190, in deco
[2025-04-25T11:53:26.105+0000] {subprocess.py:93} INFO -     return f(*a, **kw)
[2025-04-25T11:53:26.105+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^
[2025-04-25T11:53:26.105+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/protocol.py", line 326, in get_return_value
[2025-04-25T11:53:26.106+0000] {subprocess.py:93} INFO -     raise Py4JJavaError(
[2025-04-25T11:53:26.106+0000] {subprocess.py:93} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o181.isEmpty.
[2025-04-25T11:53:26.107+0000] {subprocess.py:93} INFO - : org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 0.0 failed 1 times, most recent failure: Lost task 1.0 in stage 0.0 (TID 1) (237827c6637e executor driver): org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:53:26.107+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:53:26.108+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:53:26.108+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:53:26.109+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:53:26.110+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:26.111+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:53:26.112+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:26.112+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:26.113+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:53:26.113+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:53:26.114+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:26.114+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:53:26.115+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:53:26.115+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:53:26.116+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:53:26.116+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:53:26.117+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:53:26.117+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:53:26.117+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:53:26.118+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:53:26.118+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:53:26.118+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:53:26.119+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:53:26.119+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:26.120+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:53:26.120+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:53:26.120+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:53:26.121+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:26.122+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:53:26.123+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:53:26.124+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:53:26.125+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:53:26.126+0000] {subprocess.py:93} INFO - Driver stacktrace:
[2025-04-25T11:53:26.126+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2668)
[2025-04-25T11:53:26.126+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2604)
[2025-04-25T11:53:26.127+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2603)
[2025-04-25T11:53:26.128+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-04-25T11:53:26.128+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-04-25T11:53:26.128+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-04-25T11:53:26.129+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2603)
[2025-04-25T11:53:26.129+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1178)
[2025-04-25T11:53:26.130+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1178)
[2025-04-25T11:53:26.130+0000] {subprocess.py:93} INFO - 	at scala.Option.foreach(Option.scala:407)
[2025-04-25T11:53:26.130+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1178)
[2025-04-25T11:53:26.131+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2856)
[2025-04-25T11:53:26.131+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2798)
[2025-04-25T11:53:26.131+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2787)
[2025-04-25T11:53:26.132+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2025-04-25T11:53:26.132+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:952)
[2025-04-25T11:53:26.132+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2238)
[2025-04-25T11:53:26.133+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2259)
[2025-04-25T11:53:26.133+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2278)
[2025-04-25T11:53:26.134+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:506)
[2025-04-25T11:53:26.134+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:459)
[2025-04-25T11:53:26.135+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$isEmpty$1(Dataset.scala:604)
[2025-04-25T11:53:26.135+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$isEmpty$1$adapted(Dataset.scala:603)
[2025-04-25T11:53:26.136+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:3898)
[2025-04-25T11:53:26.137+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:510)
[2025-04-25T11:53:26.138+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3896)
[2025-04-25T11:53:26.138+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)
[2025-04-25T11:53:26.139+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)
[2025-04-25T11:53:26.140+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[2025-04-25T11:53:26.140+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:53:26.141+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[2025-04-25T11:53:26.141+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3896)
[2025-04-25T11:53:26.142+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.isEmpty(Dataset.scala:603)
[2025-04-25T11:53:26.142+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-04-25T11:53:26.142+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-04-25T11:53:26.143+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-04-25T11:53:26.143+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-04-25T11:53:26.144+0000] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-04-25T11:53:26.144+0000] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2025-04-25T11:53:26.145+0000] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:282)
[2025-04-25T11:53:26.146+0000] {subprocess.py:93} INFO - 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-04-25T11:53:26.147+0000] {subprocess.py:93} INFO - 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-04-25T11:53:26.148+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.sendCommand(ClientServerConnection.java:244)
[2025-04-25T11:53:26.149+0000] {subprocess.py:93} INFO - 	at py4j.CallbackClient.sendCommand(CallbackClient.java:384)
[2025-04-25T11:53:26.150+0000] {subprocess.py:93} INFO - 	at py4j.CallbackClient.sendCommand(CallbackClient.java:356)
[2025-04-25T11:53:26.150+0000] {subprocess.py:93} INFO - 	at py4j.reflection.PythonProxyHandler.invoke(PythonProxyHandler.java:106)
[2025-04-25T11:53:26.151+0000] {subprocess.py:93} INFO - 	at jdk.proxy3/jdk.proxy3.$Proxy37.call(Unknown Source)
[2025-04-25T11:53:26.152+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1(ForeachBatchSink.scala:51)
[2025-04-25T11:53:26.153+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1$adapted(ForeachBatchSink.scala:51)
[2025-04-25T11:53:26.154+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.ForeachBatchSink.addBatch(ForeachBatchSink.scala:32)
[2025-04-25T11:53:26.154+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$17(MicroBatchExecution.scala:665)
[2025-04-25T11:53:26.155+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)
[2025-04-25T11:53:26.156+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)
[2025-04-25T11:53:26.157+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[2025-04-25T11:53:26.157+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:53:26.158+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[2025-04-25T11:53:26.158+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$16(MicroBatchExecution.scala:663)
[2025-04-25T11:53:26.158+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-25T11:53:26.159+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-25T11:53:26.159+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-25T11:53:26.160+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runBatch(MicroBatchExecution.scala:663)
[2025-04-25T11:53:26.160+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$2(MicroBatchExecution.scala:256)
[2025-04-25T11:53:26.161+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-25T11:53:26.161+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-25T11:53:26.162+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-25T11:53:26.162+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-25T11:53:26.163+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$1(MicroBatchExecution.scala:219)
[2025-04-25T11:53:26.163+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProcessingTimeExecutor.execute(TriggerExecutor.scala:67)
[2025-04-25T11:53:26.164+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runActivatedStream(MicroBatchExecution.scala:213)
[2025-04-25T11:53:26.165+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.$anonfun$runStream$1(StreamExecution.scala:307)
[2025-04-25T11:53:26.165+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-25T11:53:26.166+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:53:26.166+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.org$apache$spark$sql$execution$streaming$StreamExecution$$runStream(StreamExecution.scala:285)
[2025-04-25T11:53:26.167+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution$$anon$1.run(StreamExecution.scala:208)
[2025-04-25T11:53:26.167+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:53:26.168+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:53:26.168+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:53:26.169+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:53:26.169+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:53:26.170+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:26.171+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:53:26.171+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:26.172+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:26.172+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:53:26.173+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:53:26.174+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:53:26.174+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:53:26.175+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:53:26.175+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:53:26.176+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:53:26.176+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:53:26.177+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:53:26.177+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:53:26.178+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:53:26.178+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:53:26.178+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:53:26.179+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:53:26.179+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:53:26.180+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:26.181+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:53:26.181+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:53:26.182+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:53:26.182+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:53:26.182+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:53:26.183+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:53:26.183+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:53:26.183+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:53:26.184+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:53:26.285+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-04-25T11:53:26.306+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/bash.py", line 212, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-04-25T11:53:26.310+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=traffic_data_pipeline, task_id=submit_traffic_processor_spark_job, execution_date=20250425T115310, start_date=20250425T115312, end_date=20250425T115326
[2025-04-25T11:53:26.333+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 17 for task submit_traffic_processor_spark_job (Bash command failed. The command returned a non-zero exit code 1.; 419)
[2025-04-25T11:53:26.368+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-04-25T11:53:26.399+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
